app/controllers/api/v0/forem_directories_controller.rb

```diff
@@ -0,0 +1,27 @@
+module Api
+  module V0
+    class ForemDirectoriesController < ApiController
+      def index
```

Comment:
**blocking** this should be a `show` endpoint, it's not a list of resources, we're asking info about one singular resource: the Forem hosting this API

Reply:
@rhymes I considered this, but `show` requires an `id` which doesn't really fit this situation. What would you like to see? I'll update it right away for you.

Reply:
@rhymes I considered this, but `show` requires `id` which doesn't fit this situation - we don't have an `id` here. What would you like to see? I'll update it right away for you.

Reply:
@atsmith813 If you change the route declaration from `resources` to `resource`, the path becomes `/api/forem_directory` and the action at that path becomes `show` with no `id` requirement.

But also, that brings up another point: is the route name accurate? Is this path providing a Forem directory? Or just its own entry for that directory? I would imagine that requesting a Forem directory resource would return a list of Forem instances. The way Mastodon does this for federation and discovery is to provide [an `/api/v1/instance` route](https://github.com/mastodon/mastodon/blob/d8629e7b86b6fcaedc8b31de57dd95e85fe4fb04/config/routes.rb#L416), so the resource is the `instance`. Would a name like that make sense here?

Reply:
@jgaskins exactly, @rhymes suggested that exact change [here](https://github.com/forem/forem/pull/14301#pullrequestreview-712491593) which I'm working on. I personally prefer `index of 1` over `show`. While we can modify it to work, I've been under the impression that `show` should have `id` per convention.

I'll update it to be `instance` and work without id.

Reply:
@atsmith813 yeah, as Jamie said, if you use the singular `resource` (we have only 1 instance), you don't need the id, see https://guides.rubyonrails.org/routing.html#singular-resources

Reply:
@atsmith813 That's what I get for skimming instead of reading thoroughly. I just make a fool out of myself. :joy: